### ISSUE TITLE
Crash due to checkPermission related to internal DJI Bluetooth service

### ISSUE THREAD

**Issue by jeryini** [dji-sdk/Mobile-SDK-Android#1041]
Hello!

In my production app I'm currently seeing the following crash that happens sporadically for some users:
```
Fatal Exception: java.lang.NullPointerException
Attempt to invoke virtual method 'int android.content.Context.checkPermission(java.lang.String, int, int)' on a null object reference
androidx.core.content.ContextCompat.checkSelfPermission (ContextCompat.java:538)
dji.midware.ble.BluetoothLeService.start (Unknown Source:14)
dji.midware.data.manager.P3.ServiceManager.sfg (Unknown Source:43)
dji.midware.dgh.fdd (Unknown Source:49)
dji.sdksharedlib.fdd.fdd (Unknown Source:8)
dji.sdk.sdkmanager.DJISDKManager.initParams (Unknown Source:15)
dji.sdk.sdkmanager.DJISDKManager.initSDKManager (Unknown Source:12)
dji.sdk.sdkmanager.DJISDKManager.registerApp (Unknown Source:39)
dji.sdk.sdkmanager.DJISDKManager.access$800
dji.sdk.sdkmanager.DJISDKManager$6$1.onDownloadSuccess (Unknown Source:8)
dji.internal.ghu.ghu.fdd (Unknown Source:19)
dji.sdk.sdkmanager.DJISDKManager$6.run (Unknown Source:37)
dji.sdksharedlib.gfd.koy.run (Unknown Source:5)
```

Based on crash log it seems a call to `checkSelfPermission` is comming from internal DJI service `dji.midware.ble.BluetoothLeService.start`. This is actually quite common issue (see [this SO post](https://stackoverflow.com/questions/37322645/nullpointerexception-when-trying-to-check-permissions/37323440)) in Android where context can be null when calling `ActivityCompat.checkSelfPermission`.

Though I've moved the code related to app registration (`DJISDKManager.getInstance().registerApp`) before checking for permissions. But why is this related to Bluetooth? I don't have any permissions defined for Bluetooth, though have researched on the Internet and foud out that DJI smart controller can use Bluetooth. Is this exception perhaps due to missing required permission for Bluetooth and crash occurs if someone is using my app on the DJI smart controller?

**Comment by luoyexk** [comment]
MSDK Version：4.15.1
phones: Galaxy S20+ 5G/Galaxy S21 5G/Galaxy Note10+/Galaxy S10/Galaxy Note9 ..
drone: Mavic Air 2/Spark/Mavic 2 Pro ..
89% happened in background

Error
```
Fatal Exception: java.lang.RuntimeException: Error receiving broadcast Intent { act=android.bluetooth.adapter.action.STATE_CHANGED flg=0x4000010 (has extras) } in dji.midware.ble.BluetoothLeService$ActionReceiver@3c275b7
       at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0$LoadedApk$ReceiverDispatcher$Args(LoadedApk.java:1676)
       at android.app.-$$Lambda$LoadedApk$ReceiverDispatcher$Args$_BumDX2UKsnxLVrE6UJsJZkotuA.run(:2)
       at android.os.Handler.handleCallback(Handler.java:938)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:246)
       at android.app.ActivityThread.main(ActivityThread.java:8633)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'int android.content.Context.checkPermission(java.lang.String, int, int)' on a null object reference
       at androidx.core.content.ContextCompat.checkSelfPermission(ContextCompat.java:538)
       at dji.midware.ble.BluetoothLeService.start(:14)
       at dji.midware.ble.BluetoothLeService.onBLEOpened(:2)
       at dji.midware.ble.BluetoothLeService.access$100()
       at dji.midware.ble.BluetoothLeService$ActionReceiver.onReceive(:84)
       at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0$LoadedApk$ReceiverDispatcher$Args(LoadedApk.java:1666)
       at android.app.-$$Lambda$LoadedApk$ReceiverDispatcher$Args$_BumDX2UKsnxLVrE6UJsJZkotuA.run(:2)
       at android.os.Handler.handleCallback(Handler.java:938)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:246)
       at android.app.ActivityThread.main(ActivityThread.java:8633)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
```

**Comment by DJI-William** [comment]
@luoyexk How can you reproduce this?

**Comment by luoyexk** [comment]
> @luoyexk How can you reproduce this?

These logs are from Firebase.
I haven't seen it once in my development. So I can't reproduce it.

**Comment by jeryini** [comment]
@DJI-William I actually released a fix to check for app permissions before registration, and this still randomly happens for some users.

**Comment by DJI-William** [comment]
This issue has been fixed and merged to 4.16 branch
